pow/src/main/java/tech/pegasys/artemis/pow/DepositContractListener.java

```diff
@@ -16,19 +16,28 @@
 import static tech.pegasys.artemis.pow.contract.DepositContract.DEPOSITEVENT_EVENT;
 
 import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.UnsignedLong;
+import io.reactivex.Flowable;
 import io.reactivex.disposables.Disposable;
+import java.util.Optional;
 import org.web3j.abi.EventEncoder;
+import org.web3j.protocol.Web3j;
 import org.web3j.protocol.core.DefaultBlockParameterName;
 import org.web3j.protocol.core.methods.request.EthFilter;
+import org.web3j.protocol.core.methods.response.EthBlock;
+import org.web3j.protocol.core.methods.response.EthBlock.Block;
 import tech.pegasys.artemis.pow.contract.DepositContract;
+import tech.pegasys.artemis.pow.contract.DepositContract.DepositEventEventResponse;
 import tech.pegasys.artemis.pow.event.Deposit;
 
 public class DepositContractListener {
-
   private final Disposable subscriptionNewDeposit;
+  private final Web3j web3j;
   private DepositContract contract;
+  private volatile Optional<EthBlock.Block> cachedBlock = Optional.empty();
```

Comment:
Are we caching only the current block because the depositEventFlowable will never return an Event from a past block?

Reply:
Correct, it iterates forward only.